Flagsmith.FlagsmithClient/Flags.cs

```diff
@@ -34,17 +36,19 @@ public async Task<Flag> GetFlag(string featureName)
                 await _AnalyticsProcessor.TrackFeature(flag.GetFeatureName());
             return flag;
         }
-        public List<Flag> AllFlags() => _Flags;
+
+        public Task<IReadOnlyCollection<IFlag>> AllFlags() => Task.FromResult<IReadOnlyCollection<IFlag>>(_Flags);
+
         private static Flag FromFeatureStateModel(FeatureStateModel featureStateModel, string identityId = null) =>
```

Comment:
Since we're major versioning, I'd like to take the opportunity to clear up a few things here that I missed previously actually. For example, this (singular) FromFeatureStateModel should exist on the `Flag` class, not on the `Flags` class. 

These classes should mimic the models here: https://github.com/Flagsmith/flagsmith-python-client/blob/main/flagsmith/models.py. 

... unless there's a valid reason not to do this in C#.

Reply:
sure, np. but i would prefer to make such changes after following PR (not created yet) where are many many more things :)

Reply:
Yes, that makes sense 👍 you can leave it to me in fact if you would like to.